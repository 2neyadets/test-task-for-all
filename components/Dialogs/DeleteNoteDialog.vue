<template lang="pug">
  Dialog(:value="deleteNoteDialog" @hide="changeDialogState" @keyup.esc.native="changeDialogState")
    .delete-note-confirm
      h6.mt-none.mb-md(v-html="title" style="text-align: justify;")
      .flex.justify-end
        MyBtn(
          @click.native="changeDialogState"
          label="Отменить"
          size="16"
          flat
          autofocus
        )
        MyBtn.ml-sm(
          @click.native="deleteNote"
          label="Да, удалить"
          size="16"
        )
</template>

<script>
import { mapGetters } from 'vuex'
import Dialog from '../Reusable/Dialog'
import MyBtn from '../Reusable/MyBtn'

export default {
  name: 'DeleteNoteDialog',
  components: { MyBtn, Dialog },
  props: {},
  data () {
    return {
    }
  },
  computed: {
    ...mapGetters([
      'deleteNoteDialog',
    ]),
    ...mapGetters('notes', [
      'currentNote',
    ]),
    title () {
      return `Вы уверены, что хотите <span style='color: red;'>удалить</span> заметку с названием <strong>"${this.currentNote && this.currentNote.title}"</strong>?`
    },
  },
  watch: {},
  created () {},
  mounted () {},
  updated () {},
  methods: {
    changeDialogState () {
      this.$store.dispatch('changeDialogState', 'deleteNote')
    },
    deleteNote () {
      this.changeDialogState()
      this.$store.dispatch('notes/deleteNoteAction', this.currentNote.id)
      this.$router.push('/')
    },
  },
}
</script>
