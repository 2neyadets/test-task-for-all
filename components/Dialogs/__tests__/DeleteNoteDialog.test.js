import Vuex from 'vuex'
import VueRouter from 'vue-router'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import DeleteNoteDialog from '../DeleteNoteDialog'

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(VueRouter)
const router = new VueRouter()

describe('DeleteNoteDialog component', () => {
  let wrapper, actions, noteGetters, getters, store
  const currentNote = {
    id: 'f1733253812c-f300272d',
    title: 'Название заметки',
  }
  router.push(`/note/${currentNote.id}`)

  beforeEach(() => {
    actions = {
      changeDialogState: jest.fn(),
      'notes/deleteNoteAction': jest.fn(),
    }
    noteGetters = {
      currentNote: () => ({ ...currentNote }),
    }
    getters = {
      deleteNoteDialog: () => true,
    }
    store = new Vuex.Store({
      actions,
      getters,
      modules: {
        notes: {
          namespaced: true,
          getters: noteGetters,
        },
      },
    })

    wrapper = shallowMount(DeleteNoteDialog, {
      localVue,
      store,
      router,
    })
  })

  it('mounts without errors', done => {
    localVue.config.errorHandler = done // для отлавливания ошибок внутри nextTick, если нужно что-то выпонять внутри него
    expect.assertions(1)
    expect(wrapper).toBeTruthy()
    done()
  })

  it('can close dialog', () => {
    expect.assertions(1)
    const btn = wrapper.findAll('MyBtn-stub').at(0)
    btn.trigger('click')
    expect(actions.changeDialogState).toHaveBeenCalled()
  })

  it('can delete note', () => {
    expect.assertions(3)
    const btn = wrapper.findAll('MyBtn-stub').at(1)
    btn.trigger('click')
    expect(actions.changeDialogState).toHaveBeenCalled()
    expect(actions['notes/deleteNoteAction']).toHaveBeenCalled()
    expect(router.currentRoute.fullPath).toBe('/')
  })

  it('contains note.title', () => {
    expect.assertions(1)
    expect(wrapper.find('H6').html()).toContain(currentNote.title)
  })

  it('match snapshot', () => {
    expect.assertions(1)
    expect(wrapper.element).toMatchSnapshot()
  })
})
