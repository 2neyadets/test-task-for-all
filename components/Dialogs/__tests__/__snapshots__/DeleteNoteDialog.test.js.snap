// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteNoteDialog component match snapshot 1`] = `
<dialog-stub
  value="true"
>
  <div
    class="delete-note-confirm"
  >
    <h6
      class="mt-none mb-md"
      style="text-align: justify;"
    >
      Вы уверены, что хотите 
      <span
        style="color: red;"
      >
        удалить
      </span>
       заметку с названием 
      <strong>
        "Название заметки"
      </strong>
      ?
    </h6>
    <div
      class="flex justify-end"
    >
      <mybtn-stub
        autofocus="true"
        flat="true"
        icon=""
        label="Отменить"
        size="16"
        type="button"
      />
      <mybtn-stub
        class="ml-sm"
        icon=""
        label="Да, удалить"
        size="16"
        type="button"
      />
    </div>
  </div>
</dialog-stub>
`;
